hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/BlockID.java

```diff
@@ -120,13 +157,13 @@ public boolean equals(Object o) {
     }
     BlockID blockID = (BlockID) o;
     return containerBlockID.equals(blockID.getContainerBlockID())
-        && blockCommitSequenceId == blockID.getBlockCommitSequenceId();
+        && blockCommitSequenceId == blockID.getBlockCommitSequenceId()
+        && Objects.equals(replicaIndex, blockID.getReplicaIndex());
```

Comment:
What does Objects.equals() do if both arguments are "null"?

Reply:
Looks like Objects.equals handles it as you would expect:

> If both objects are null it will return true, if one is null and another isn't it will return false. Otherwise it will return the result of calling [equals](http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals%28java.lang.Object%29) on the first object with the second as argument.

Reply:
yeah